src/main/java/net/draycia/carbon/listeners/contexts/EconomyContext.java

```diff
@@ -19,14 +20,18 @@ public EconomyContext(@NonNull final CarbonChat carbonChat) {
     this.economy = this.carbonChat.getServer().getServicesManager().getRegistration(Economy.class).getProvider();
 
     CarbonEvents.register(PreChatFormatEvent.class, event -> {
-      final Object requiredBalObject = event.channel().context("vault-balance");
+      final Context context = event.channel().context("vault-balance");
+
+      if (context == null) {
+        return;
+      }
 
       final Double requiredBal;
 
-      if (requiredBalObject instanceof Double) {
-        requiredBal = (Double) requiredBalObject;
-      } else if (requiredBalObject instanceof Integer) {
-        requiredBal = Double.valueOf((Integer) requiredBalObject);
+      if (context.isDouble()) {
```

Comment:
Could we add a `#isNumber` and `#asNumber` as `Integer`, `Double`, and more extend it? Thereby we could just `context.isNumber()` & `context.asNumber().doubleValue()`.

Reply:
Sounds good, will do tonight after testing.